.github/scripts/trymerge.py

```diff
@@ -1291,6 +1291,16 @@ def merge(pr_num: int, repo: GitRepo,
 
     check_for_sev(org, project, force)
 
+    gh_post_pr_comment(org, project, pr.pr_num, explainer.get_merge_message(land_check_commit))
+    if (datetime.utcnow() - pr.last_pushed_at()).days > stale_pr_days:
+        raise RuntimeError("This PR is too stale; the last push date was more than 3 days ago. Please rebase and try again.")
+
+    # Important: check for merge rule once before starting land checks
+    # because we want to make sure that only approved PRs can start CI
+    # jobs. If there's missing approval, a RuntimeError will be raised
+    # here to stop the merge process right away
+    find_matching_merge_rule(pr, repo, force=True)
```

Comment:
nit: feel like we should rename the force param cause we're not actually forcing it. Also can we move this down to the if land_checks: ? If we make changes to this function, we might unintentionally break force merge on lines 1304-1307